src/module.ts

```diff
@@ -2409,22 +2409,31 @@ export class Module {
     let cStr1 = this.allocStringCached(exportName);
     let cStr2 = this.allocStringCached(name);
     let k = segments.length;
-    let segs = new Array<usize>(k);
+    let empty = 0;
+
+    let segs = new Array<usize>();
     let psvs = new Uint8Array(k);
-    let offs = new Array<ExpressionRef>(k);
-    let sizs = new Array<Index>(k);
-    for (let i = 0; i < k; ++i) {
+    let offs = new Array<ExpressionRef>();
+    let sizs = new Array<Index>();
+    for (let i = 0; i < k; i++) {
       let segment = unchecked(segments[i]);
       let buffer = segment.buffer;
+      if (!buffer) {
+        empty++;
+        continue;
+      }
+
       let offset = segment.offset;
-      unchecked(segs[i] = allocU8Array(buffer));
-      unchecked(psvs[i] = 0); // no passive segments currently
-      unchecked(offs[i] = target == Target.Wasm64
+      segs.push(allocU8Array(buffer));
+      psvs[i - empty] = 0; // no passive segments currently
```

Comment:
`psvs` is always 0, then we only need to create a U8Array at the end and set it to zero.

Reply:
I kept it there in case we do something different in the future. The `unchecked(psvs[i] = 0);` line didn't *need* to be there anyway, just as `psvs[i - empty] = 0;` doesn't *need* to be here now.